### PR TITLE
Update juju filesystem state from k8s

### PR DESCRIPTION
**Pull request by wallyworld**
## Description of change

k8s unit updates to juju now contain filesystem/volume information which juju uses to updates its view of the units' filesystemm attachments.

A fix was also made to k8s statefullset provisioning - we need to ensure the volume claim template is not updated once the set is created.

The major bit are:
- k8s client gathers filesystem and volume info
- caasunitprovisioner worker sends changes back to facade
- caasunitprovisioner facade updates state with new filesystem info
- new Containers() api in state to allow unit tags to be queried

## QA steps

Deploy a mysql k8s charm with storage, run
juju storage
juju storage --filesystem
the above will show the filesystem becoming attached

### CODE REVIEW

File: caas/kubernetes/provider/k8s.go [hunk 1]
```diff
@@ -489,21 +490,22 @@ func (k *kubernetesClient) configureStorage(
...
+		pvcNamePrefix := fs.Tag.String()
 		volStorageLabel := fmt.Sprintf("%s-unit-storage", appName)
 		params := volumeParams{
-			storageLabels: []string{volStorageLabel, k.namespace, "default"},
-			pvcName:       pvcName,
-			volumeSize:    fmt.Sprintf("%dMi", fs.Size),
-			labels:        map[string]string{labelApplication: appName},
+			storageLabels:       []string{volStorageLabel, k.namespace, "default"},
+			pvcName:             pvcNamePrefix,
+			requestedVolumeSize: fmt.Sprintf("%dMi", fs.Size),
```
Comment (ycliuhw) [inline caas/kubernetes/provider/k8s.go]: do we only allow `Mi`?

Comment (wallyworld) [reply caas/kubernetes/provider/k8s.go]: The Juju state model stores the required sizes in MiB

Comment (ycliuhw) [reply caas/kubernetes/provider/k8s.go]: ic, thx

**Comment by ycliuhw** [comment]
LGTM thanks.

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**